handlers/handlers.go

```diff
@@ -17,25 +18,28 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-type StreamInfo struct {
-	callbackUrl string
-}
-
 type CatalystAPIHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
 func (d *CatalystAPIHandlersCollection) Ok() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
 		io.WriteString(w, "OK")
 	}
 }
 
-func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
+// Takes mpegts segment as input, ingests in Mist under SOURCE_PREFIX streamName, transcodes into RENDITION_PREFIX stream then pushes renditions to specified destination.
+// Manually starts MistLivepeerProc binary. For this we require port of B-node that listens on 127.0.0.1 .
+// When `MistProcLivepeer` starts, SOURCE_PREFIX stream is taken as input and (will automatically - WIP) start SOURCE_PREFIX stream.
+// No trigger is registered on SOURCE_PREFIX so configured sourceUrl is used.
+// Also `MistProcLivepeer` outputs to RENDITION_PREFIX stream where we listen for triggers to start new push to rendition destination.
+// Because RENDITION_PREFIX stream contains multiple video tracks we start multiple push-es each selecting one video and one audio track to push to S3.
+func (d *CatalystAPIHandlersCollection) TranscodeSegment(broadcasterPort int, mistProcPath string) httprouter.Handle {
 	schema := inputSchemasCompiled["TranscodeSegment"]
 
 	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```

Comment:
For the sake of readability, could you try to split this function into multiple functions? Or at least extract a few functions from it? E.g. the logic related to starting the `MistProcLivepeer` could be moved to a separate function.

Reply:
Great comment. I really prefer splitting into multiple functions.